python/triton/runtime/interpreter.py

```diff
@@ -703,7 +780,12 @@ def _get_bool(self):
         return bool(data) if data.size == 1 else True
 
     def _get_transpose(self):
-        return tl.core.tensor(TensorHandle(np.transpose(self.handle.data), self.handle.dtype), self.dtype.scalar)
+        handle = TensorHandle(np.transpose(self.handle.data), self.handle.dtype)
+        assert self.type.is_block()
+        block_shape = list(self.type.shape)
+        block_shape[-1], block_shape[-2] = block_shape[-2], block_shape[-1]
+        res_ty = tl.core.block_type(self.dtype, block_shape)
+        return tl.core.tensor(handle, res_ty)
```

Comment:
Had to change this because `tl.dot(a, b.T, accumulator)` was failing `assert rhs.is_block()`. The previous implementation uses `self.dtype.scalar` which is not a block type.

Reply:
Indeed yes. Thanks for the fix